### PR TITLE
Updates the asset pipeline lesson for Rails 6

### PR DESCRIPTION
**Pull request by CouchofTomato**
Rails 6 doesn't use the asset pipeline by default any longer for javascript files.

This commit updates the asset pipeline lesson to explain this but does keep content for how the asset pipeline can handle javascript files as you can still opt to use the asset pipeline to handle them

### CODE REVIEW

**Review by KevinMulhern — APPROVED** [review]
:shipit:

**Review by KevinMulhern — COMMENTED** [review]
Looks ace @CouchofTomato, thanks for putting all of this together. Just a few minor nitpicks.

File: rails_programming/rails_basics/webpacker.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+### Introduction
+
+The Asset Pipeline was a great feature for Rails. It made it easy to write CSS and Javascript and let the Asset Pipeline do the heavy lifting of bundling all your stylesheets together and it did the same with Javascript. It wasn't without drawbacks though. Often it took time to integrate the latest Javascript language features and gems that loading in popular Javascript libraries and frameworks weren't always at the cutting edge. This gave Rails a reputation for being reliable but outdated. Especially with big strides being made in the Javascript community of creating an ecosystem that was much easier to work with than it had a reputation for in the past.
```
[the remark this answers is not in the document]
Comment (CouchofTomato) [reply rails_programming/rails_basics/webpacker.md]: I realised it was a bit harsh to refer to the asset pipeline as something that used to be great when it still is so refactored along those lines.

File: rails_programming/rails_basics/webpacker.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Webpack was created as a static module bundler for modern Javascript applications which made it much easier to work with ensuring the code you wrote in development to be easy to read and work with could be processed into something better suited for browsers when code was moved to production. More importantly Webpack could work with the node package manager, giving access to a huge library of Javascript libraries and as soon as they were released you could update.
```
[the remark this answers is not in the document]
Comment (CouchofTomato) [reply rails_programming/rails_basics/webpacker.md]: No, I just worded it terribly.

File: rails_programming/rails_basics/webpacker.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+This is one key difference between the Asset Pipeline and Webpacker. With the Asset Pipeline all Javascript code was bundled into one large file, with the small exception of code used to execute on Ajax requests which could be executed on at the time the request came in, and so all files had to be referenced in the `application.js` file. That file was downloaded to a client when it first connected to your server. Using packs instead, we can have different packs that can be downloaded only when needed. You can create an unlimited number of packs and just reference that pack on the page it is needed. For example if you had some code to handle form validations on the client side it might not be needed for somebody who never went to the page with the form. With the Asset Pipeline that code would be downloaded at initial page load but using Webpack you can just reference that pack in the view where it is required and it will only download when somebody visits the form.
```
[the remark this answers is not in the document]
Comment (CouchofTomato) [reply rails_programming/rails_basics/webpacker.md]: 👍

**Review by KevinMulhern — APPROVED** [review]
Brilliant @CouchofTomato, thanks for doing this.

**Pull request merged**